[Timber] update a11y headings and remove stripe icon whitelist

matcaissy opened the pull request:
Fix for https://github.com/Shopify/shopify-themes/issues/3527 (improving theme accessibility) and removed Stripe in credit cart payment whitelist.

**Demo: [link](http://timber-a11y.myshopify.com/?key=a0335d17c5dc8dbd88dc16199fc42229156bf8b8ba253d9d26aceb39170de2a4&preview_theme_id=110651457)**

cc @cshold @schaeken 👀

cshold commented:
Not sure the changes to heading tags are needed in most of these cases. They should reflect the importance of the header on the page, meaning an `h2` is the second most important thing. "Recent articles" and "Blog categories" in the sidebar don't fit that, for example.

schaeken commented:
What is more important than "recent articles" on the blog page. The blog title? That's an `h1`. The footer headings are `h3`s. So the "recent articles" heading kinda makes sense as an `h2` but are we assuming merchants don't use headings in their blog content?

I really hope the new RTE will have one type of heading or at the most a subheading for the merchant to use. Then they will input all the `h2`s and `h3`s in the theme and we can put our headings into lower priority. Maybe we should leave `h2`s for the merchant?

matcaissy commented:
Thanks for the feedback @cshold @schaeken. I began looking at article.liquid so updating the sidebar headings from h4 to h2 made sense there:

![https://screenshot.click/mtkw6-mblq7.png](https://screenshot.click/mtkw6-mblq7.png)

Since it's the same snippet, it has to also work on blog.liquid

![https://screenshot.click/d7fbq-rf5t7.png](https://screenshot.click/d7fbq-rf5t7.png)

How about changing to a `h3`? Recent Articles `h3` in this example. Looks better when the comments are added:

![https://screenshot.click/2g1bv-afzch.png](https://screenshot.click/2g1bv-afzch.png)

matcaissy commented:
**Updated**: blog sidebar headings to h3

matcaissy commented:
@cshold I have updated blog tags and categories to `h3` in the `blog-sidebar.liquid` which makes more sense for Timber. Should we leave collection tags as `h3`? For now, we have updated them as h2 in this PR, and as well in [Supply - see commit](https://github.com/Shopify/shopify-themes/pull/3438/files#diff-62ea3b838a35b473d682cb6995b7fc85R19).

![https://screenshot.click/rofqy-yu8z0.png](https://screenshot.click/rofqy-yu8z0.png)

Let us know - we can revert to `h3` in Supply. cc/ @jeremystephencobb

cshold commented:
Looks good to me. 🚢 anytime
